### PR TITLE
Adding Edit disk modal

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

Adding Edit disk modal to VM disks tab.

## 🎥 Demo

https://user-images.githubusercontent.com/67270715/160815628-bdcc95a2-bb3c-4f88-8deb-735e95056d7f.mp4

https://user-images.githubusercontent.com/67270715/160815638-7da68443-91fc-42a0-b30a-45076da06ebf.mp4

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

File: src/utils/components/DiskModal/EditDiskModal.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  const [diskSourceState, dispatchDiskSourceState] = React.useReducer(
+    diskSourceReducer,
+    initialDiskSourceState,
+  );
+  const sourceRequiresDataVolume = React.useMemo(
+    () => requiresDataVolume(diskState.diskSource),
+    [diskState.diskSource],
+  );
+
+  const updatedVirtualMachine: V1VirtualMachine = React.useMemo(() => {
```
Comment (upalatucci) [inline src/utils/components/DiskModal/EditDiskModal.tsx:59]: Can we split this function into smaller ones and move them into a utils file?

**Review by upalatucci — COMMENTED** [review]
:+1:

File: src/utils/components/DiskModal/EditDiskModal.tsx (hunk 1 above)
Comment (avivtur) [reply src/utils/components/DiskModal/EditDiskModal.tsx:64]: fixed thanks :)

**Pull request merged**